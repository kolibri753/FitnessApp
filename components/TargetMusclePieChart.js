import React from "react";
import { View, Text, StyleSheet } from "react-native";
import PieChart from "react-native-pie-chart";
import { targetColors } from "../styles/colors";

const TargetMusclePieChart = ({ targets }) => {
	const uniqueTargets = Array.from(new Set(targets));
	const targetCounts = uniqueTargets.map(
		(target) => targets.filter((t) => t === target).length
	);

	const sliceColor = uniqueTargets.map(
		(target) => targetColors[target] || "#FFFF00"
	);

	return (
		<View style={styles.container}>
			<Text style={styles.title}>Target Muscles</Text>
			<PieChart
				widthAndHeight={250}
				series={targetCounts}
				sliceColor={sliceColor}
			/>
			<View style={styles.legendContainer}>
				{uniqueTargets.map((target, index) => (
					<View key={index} style={styles.legendItem}>
						<View
							style={[
								styles.legendColor,
								{ backgroundColor: targetColors[target] || "#FFFF00" },
							]}
						/>
						<Text style={styles.legendText}>{target}</Text>
						<Text style={[styles.legendPercentage, { color: targetColors[target] || "#FFFF00" }]}>
							{((targetCounts[index] / targets.length) * 100).toFixed(2)}%
						</Text>
					</View>
				))}
			</View>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		alignItems: "center",
		color: "#fff",
	},
	title: {
		fontSize: 18,
		fontWeight: "bold",
		marginBottom: 10,
		color: "#fff",
	},
	legendContainer: {
		flexDirection: "row",
		flexWrap: "wrap",
		justifyContent: "center",
		marginVertical: 10,
	},
	legendItem: {
		flexDirection: "row",
		alignItems: "center",
		marginRight: 20,
		marginBottom: 5,
	},
	legendColor: {
		width: 20,
		height: 20,
		borderRadius: 10,
		marginRight: 5,
	},
	legendText: {
		fontSize: 16,
		marginRight: 5,
		color: "#fff",
	},
	legendPercentage: {
		fontSize: 18,
	},
});

export default TargetMusclePieChart;
