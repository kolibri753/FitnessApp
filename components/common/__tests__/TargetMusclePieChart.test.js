import React from "react";
import { render } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import TargetMusclePieChart from "../TargetMusclePieChart";

jest.mock("react-native-pie-chart", () => "PieChartMock");

describe("TargetMusclePieChart", () => {
	it("renders title correctly", () => {
		const { getByText } = render(<TargetMusclePieChart targets={[]} />);
		const title = getByText("Target Muscles");
		expect(title).toBeTruthy();
	});

	it("renders 'Add exercises' text when no data is provided", () => {
		const { getByText } = render(<TargetMusclePieChart targets={[]} />);
		const noDataText = getByText("Add exercises to this workout!");
		expect(noDataText).toBeTruthy();
	});

	it("renders legend items correctly", () => {
		const targets = ["Legs", "Arms", "Back"];
		const { getByText } = render(<TargetMusclePieChart targets={targets} />);

		targets.forEach((target) => {
			const legendText = getByText(target);
			expect(legendText).toBeTruthy();
		});
	});

  it("renders correctly", () => {
		const tree = renderer.create(<TargetMusclePieChart />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
