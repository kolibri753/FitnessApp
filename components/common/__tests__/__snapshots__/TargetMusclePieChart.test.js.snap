// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetMusclePieChart renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "color": "#fff",
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Target Muscles
  </Text>
  <Text
    style={
      {
        "color": "#fff",
      }
    }
  >
    Add exercises to this workout!
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginVertical": 10,
      }
    }
  />
</View>
`;
